source/interprocedural_analyses/taint/taintConfiguration.ml

```diff
@@ -780,8 +794,21 @@ module Error = struct
           sink
           (String.concat labels ~sep:", ")
     | InvalidMultiSink sink -> Format.fprintf formatter "`%s` is not a multi sink" sink
-    | RuleCodeDuplicate code ->
-        Format.fprintf formatter "Multiple rules share the same code `%d`" code
+    | RuleCodeDuplicate { code; previous_location = None } ->
+        Format.fprintf
+          formatter
+          "Multiple rules share the same code `%d`, previous rule was defined inside the pyre \
+           binary"
```

Comment:
That would be confusing if we showed that to the user. They wouldn't know what the "pyre binary" is.
I guess this can only happen in tests right?
```suggestion
          "Multiple rules share the same code `%d`"
```

Reply:
No, location is `None` for rules defined implicitly as I mentioned before:

> Not sure if we make it non-optional what to do for implicitly defined rules. For example, consider these defined in taintConfiguration.ml, which position and path should we use?
> 
> https://github.com/facebook/pyre-check/blob/8c639dca3a937026cf6d14130964cc485923c066/source/interprocedural_analyses/taint/taintConfiguration.ml#L515-L610

Maybe we can do something like: `Rule code `%d` cannot be used as it is reserved internally by pysa. Please use a different rule code instead`

Reply:
Those rules are in `TaintConfiguration.default`, which should only be used in tests, right?

Reply:
Yes you are right. They are only used in tests. We can go with the above suggested message in that case :)